### DISCUSSION TITLE
Is there a Way to send a Post-Request with the select-Component

### DISCUSSION THREAD

**Discussion by Adadlp** [formio/formio.js#5279] [Q&A] [answered]
Is there a way to send a Post-Request instead of a Get-Request for the Select-Component with URL-Select? Alternativ is there a way to to get the search value of the component in the calculated value Function of the Component?

**Comment by brendanbond** [comment, marked as answer]
> Is there a way to send a Post-Request instead of a Get-Request for the Select-Component with URL-Select?

I don't believe there is a way to send a POST request out of the box without extending the component itself, although there's nothing stopping, say, a Custom Default Value or a Calculated Value field from executing a custom fetch request. Additionally, our enterprise offering has some components that are able to execute complex queries and store the data in your form's data model right out of the box.

> Alternativ is there a way to to get the search value of the component in the calculated value Function of the Component?

I'm not sure what you mean by "get the search value of the component," can you be more clear?

**Reply by Adadlp** [reply]
I mean that if you use URL as data source then the value in the searchbar can be added to the Request URL. Can one get that value from the searchbar to use it in the calculated Value Field for a custom fetch request or can it be used in the custom data source?

**Reply by brendanbond** [reply]
Assuming you have selected "ChoicesJS" as the select component's driver, you may be able to hook into the underlying ChoicesJS instance and add e.g. [an event listener for "search,"](https://github.com/Choices-js/Choices#events) although I've never tried this

**Reply by Adadlp** [reply]
Ok, i will try that
